casual/casual-test/src/main/java/se/laz/casual/test/CasualEmbeddedServer.java

```
@@ -0,0 +1,217 @@
+ *
+ * NB - Currently provides just a casual event server interface.
+ */
+public class CasualEmbeddedServer
+{
+    private final UUID domainId;
+    private final Integer eventServerPort;
+    private final boolean eventServerEnabled;
+    private AtomicBoolean running = new AtomicBoolean( false );
+    private AtomicBoolean eventServerRunning = new AtomicBoolean( false );
+    private EventServer eventServer;
+
+    private CasualEmbeddedServer( Builder builder )
+    {
+        this.domainId = UUID.randomUUID();
+        this.eventServerPort = builder.eventServerPort;
+        this.eventServerEnabled = builder.eventServerEnabled;
+    }
+
+    /**
+     * Get the port for the event server instance.
+     *
+     * @return optional populated with port if provided otherwise empty.
+     */
+    public Optional<Integer> getEventServerPort()
+    {
+        return Optional.ofNullable( eventServerPort );
+    }
+
+    /**
+     * Is the event server enabled?
+     *
+     * @return if event server is enabled.
+     */
+    public boolean isEventServerEnabled()
+    {
+        return eventServerEnabled;
+    }
+
+    /**
+     * Is the event server currently running?
+     *
+     * @return if event server is running.
+     */
+    public boolean isEventServerRunning()
+    {
+        return eventServerRunning.get();
+    }
+
+    /**
+     * Is the casual embedded server running?
+     *
+     * @return if the embedded server is running.
+     */
+    public boolean isRunning()
+    {
+        return running.get();
+    }
+
+    /**
+     * Start the casual embedded server.
+     *
+     * NB - can only be called when the server is not running.
+     */
+    public void start()
+    {
+        if( running.compareAndSet( false, true ) )
+        {
+            initialiseEventServer();
+            return;
+        }
+        throw new IllegalStateException( "Server is already started." );
```

would it not be ok for start to be idempotent?

---

It is idempotent. Idempotency is about the resulting state of the system not the response. 
If you have told it to start multiple times, you have messed up i believe, hence illegal state.

---

After discussion we agreed it was ok to fail fast on multiple starts, as it is more likely user error than desired.